core/aws-core/src/main/java/software/amazon/awssdk/awscore/internal/AwsExecutionContextBuilder.java

```diff
@@ -350,11 +353,23 @@ private static EndpointProvider resolveEndpointProvider(SdkRequest request,
                       .orElse(clientConfig.option(SdkClientOption.ENDPOINT_PROVIDER));
     }
 
-    private static BusinessMetricCollection resolveUserAgentBusinessMetrics(SdkClientConfiguration clientConfig) {
+    private static <InputT extends SdkRequest, OutputT extends SdkResponse> BusinessMetricCollection
+        resolveUserAgentBusinessMetrics(SdkClientConfiguration clientConfig, 
+                                        ClientExecutionParams<InputT, OutputT> executionParams) {
         BusinessMetricCollection businessMetrics = new BusinessMetricCollection();
         Optional<String> retryModeMetric = resolveRetryMode(clientConfig.option(RETRY_POLICY),
                                                             clientConfig.option(RETRY_STRATEGY));
         retryModeMetric.ifPresent(businessMetrics::addMetric);
+
+        if (isRpcV2CborProtocol(executionParams.getProtocolMetadata())) {
```

Comment:
Just to double check, rpcv2cbor is resolved per client, not per request, right? i.e., the SDK applies cbor protocol for all operations instead of per operation